src/System.Private.TypeLoader/src/Internal/Reflection/Execution/AssemblyBinderImplementation.Ecma.cs

```diff
@@ -117,27 +174,22 @@ public PEInfo(RuntimeAssemblyName name, MetadataReader reader, PEReader pe)
                         // Not found in already loaded list, attempt to source assembly from disk
                         foreach (string filePath in FilePathsForAssembly(refName))
                         {
-                            FileStream ownedFileStream = null;
-                            PEReader ownedPEReader = null;
+                            PEReader? ownedPEReader = null;
                             try
                             {
                                 if (!RuntimeAugments.FileExists(filePath))
                                     continue;
 
                                 try
                                 {
-                                    ownedFileStream = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.Read);
+                                    ownedPEReader = OpenPEFile(filePath, out var memoryMappedView);
```

Comment:
Pass `memoryMappedView` into the `new PEInfo` constructor below, and also `Dispose()` it when it gets unused?

Reply:
Ah good catch!